connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -77,9 +77,7 @@ message Expression {
       int32 year_month_interval = 20;
       int64 day_time_interval = 21;
 
-      Array array = 22;
-      Struct struct = 23;
-      Map map = 24;
```

Comment:
We do support array literals but that's done at the client side (combination of array of literals) to have the consistent type coercion rule.

Map and structs are not supported yet in Python side because it's arguable for `tuple` and `dict` to be `struct` and `struct` as both can be read as `array` and `map` in `createDataFrame`.